src/uct/api/uct.h

```diff
@@ -385,6 +385,16 @@ enum {
                                                sockaddr */
 };
 
+/*
+ * @ingroup UCT_MD
+ * @brief  Memory types
+ */
+typedef enum {
+    UCT_MD_MEM_TYPE_DEFAULT = 0,   /**< Default system memory */
+    UCT_MD_MEM_TYPE_CUDA,          /**< NVIDIA CUDA memory */
```

Comment:
s/CUDA/GPU ?

I don't think it is a good idea to go with a very specific naming.

Reply:
@shamisp the reason for this to differentiate multiple GPU Types (nvidia, amd, ..etc)

Reply:
I don't like the usage of vendor specific naming in the API. It supposed to be abstracted in some way.

Reply:
@shamisp I can change this to s/CUDA/GPU given that there may not be heterogeneous GPU support.  one another way is to enumerate (GPU_TYPE_0, GPY_TYPE_1..).

Reply:
Yes, we can create more than one type of GPUs. @yosefe your take ?

Reply:
i think it should be "CUDA", since the user would pass to this API memory buffers allocated by cuda library. which should be distinguished from passing pointers allocated by rocm library. 
So the reason to have this is because user application already has to be aware of the memory type.

Reply:
It the enum is expected to be exposed in UCP ?

Reply:
@shamisp  we may need to know no.of  types.   but not the enum values directly.

Reply:
@shamisp i think we should, at least as flags for ucp_init() - which memory types should UCX support. This way applications/libraries can enable/disable support for GPU buffers

Reply:
I think, I can leave with CUDA memory at UCT layer, but I don't think it has to be exposed as "CUDA" in UCP layer.
And yes, we have to add something to init to give a hint about GPU usage.

Reply:
I agree with @shamisp that it will be better if we can avoid the types in UCP. but at the same time I'm wondering how UCP can select which UCT to use if it does not know the exact type. Because I can have CUDA installed in ROCM node which means I can configure UCX with both ROCM and CUDA support

Reply:
@khamidouche agreed. I'm not sure how the detection will work. It is reasonable to assume that some system will have both types of GPUs.